core/arch/aarchxx/mangle.c

```diff
@@ -603,6 +603,9 @@ insert_push_all_registers(dcontext_t *dcontext, clean_call_info_t *cci,
                                   DR_REG_LIST_LENGTH_ARM, DR_REG_LIST_ARM));
     }
     dstack_offs += 15 * XSP_SZ;
+
+    /* Make dstack_offs 8-byte algined, as we only accounted for 31 4-byte slots */
```

Comment:
I'm not sure where your 31 comes from?  I see 15 here and 2 above for pc and flags, for 17 total beyond the SIMD.

Reply:
Yes right, not sure, the 31 4-byte slot are wrong, it should be 17 4-byte slots.